pkg/theiactl/commands/start.go

```diff
@@ -0,0 +1,378 @@
+				Driver: sparkv1.DriverSpec{
+					CoreRequest: &driverCoreRequest,
+					SparkPodSpec: sparkv1.SparkPodSpec{
+						Memory: &driverMemory,
+						Labels: map[string]string{
+							"version": sparkVersion,
+						},
+						EnvSecretKeyRefs: map[string]sparkv1.NameKey{
+							"CH_USERNAME": {
+								Name: "clickhouse-secret",
+								Key:  "username",
+							},
+							"CH_PASSWORD": {
+								Name: "clickhouse-secret",
+								Key:  "password",
+							},
+						},
+						ServiceAccount: ConstStrToPointer(sparkServiceAccount),
+					},
+				},
+				Executor: sparkv1.ExecutorSpec{
+					CoreRequest: &executorCoreRequest,
+					SparkPodSpec: sparkv1.SparkPodSpec{
+						Memory: &executorMemory,
+						Labels: map[string]string{
+							"version": sparkVersion,
+						},
+						EnvSecretKeyRefs: map[string]sparkv1.NameKey{
+							"CH_USERNAME": {
+								Name: "clickhouse-secret",
+								Key:  "username",
+							},
+							"CH_PASSWORD": {
+								Name: "clickhouse-secret",
+								Key:  "password",
+							},
+						},
+					},
+					Instances: &sparkResourceArgs.executorInstances,
+				},
+			},
+		}
+		response := &sparkv1.SparkApplication{}
+		err = clientset.CoreV1().RESTClient().
+			Post().
+			AbsPath("/apis/sparkoperator.k8s.io/v1beta2").
+			Namespace(flowVisibilityNS).
+			Resource("sparkapplications").
+			Body(recommendationApplication).
+			Do(context.TODO()).
+			Into(response)
+		if err != nil {
+			return err
+		}
+		fmt.Printf("A new policy recommendation job is created successfully, id is %s\n", recommendationID)
+		return nil
+	},
+}
+
+func init() {
+	policyrecoCmd.AddCommand(startCmd)
+	startCmd.Flags().StringP(
+		"type",
+		"t",
+		"initial",
+		"{initial|subsequent} Indicates this recommendation is an initial recommendion or a subsequent recommendation job.",
+	)
+	startCmd.Flags().IntP(
+		"limit",
+		"l",
+		0,
+		"The limit on the number of flow records read from the database. 0 means no limit.",
+	)
+	startCmd.Flags().IntP(
+		"option",
+		"o",
+		1,
+		`{1|2|3} Option of network isolation preference in policy recommendation.
+Currently we support 3 options:
+1: Recommending allow ANP/ACNP policies, with default deny rules only on applied to Pod labels which have allow rules recommended.
+2: Recommending allow ANP/ACNP policies, with default deny rules for whole cluster.
+3: Recommending allow K8s network policies, with no deny rules at all`,
+	)
+	startCmd.Flags().StringP(
+		"start_time",
+		"s",
+		"",
+		`The start time of the flow records considered for the policy recommendation.
+Format is YYYY-MM-DD hh:mm:ss in UTC timezone. No limit of the start time of flow records by default.`,
+	)
+	startCmd.Flags().StringP(
+		"end_time",
+		"e",
+		"",
+		`The end time of the flow records considered for the policy recommendation.
+Format is YYYY-MM-DD hh:mm:ss in UTC timezone. No limit of the end time of flow records by default.`,
+	)
+	startCmd.Flags().StringP(
+		"ns_allow_list",
```

Comment:
In K8S' world don't CLI option use a dash as a separator?

Reply:
Right, I changed to use dashes for the flags in CLI.